code/test/Templates.Test/GenTests/GenProjectsTests.cs

```diff
@@ -48,6 +49,59 @@ public GenProjectTests(GenerationFixture fixture)
 
         [Theory]
         [MemberData("GetProjectTemplatesForGenerationAsync")]
+        [Trait("Type", "GenerationProjects")]
+        public async Task GenEmptyProjectCorrectInfeProjectConfigInfoAsync(string projectType, string framework, string language)
```

Comment:
What does this test do? The name is not clear and appears to contain a typo.

Reply:
When a solution is generated we add some metadata to the manifest to identify the framework and project type. If this information is missing, we try to infer it (there is a typo in the method name it should be "Infer" instead of "Infe") and add the correct tags to the manifest. This is done by the ProjectConfigInfo class.

Now, every generation test ensures we are adding the expected tags adding [a new assertion](https://github.com/Microsoft/WindowsTemplateStudio/pull/1169/files/77d09d84e02866747761799174745127c09feebd#diff-4b6d07657a0d2ccdc4eec08d69b11e50R58), that is validating the information generated matches with what is expected. 

In particular, this tests forces to make the infering removing the generated tags and calling the ProjectConfigInfo class. Then validates that the re-generated tags are as expected. This was implemented to have backward compatibility with project generated before right-click exists. And it is true that maybe there should not be too much projects created without the metatada, but just in case, we think it is interesting to validate that the inferention algorithm works in every case.